### PR TITLE
Modernization

### PR DESCRIPTION
**Pull request by gounthar**
Following #136 , I added a few more steps in the modernization.
Unfortunately, they don't solve the `getJenkins() is null` problem, but they add a few things (spotbugs, dependabot), following the [tutorial](https://www.jenkins.io/doc/developer/tutorial-improve/).

`20:35:07  java.lang.NullPointerException: Cannot invoke "org.jvnet.hudson.test.JenkinsRule.createProject(java.lang.Class)" because the return value of "io.jenkins.plugins.util.IntegrationTest.getJenkins()" is null
`
### Testing done

`mvn clean verify
`
### Submitter checklist
- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- ~~Link to relevant issues in GitHub or Jira~~
- [X] Link to relevant pull requests, esp. upstream and downstream changes
- ~~Ensure you have provided tests - that demonstrates feature works or fixes the issue~~

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+version: 2
```
Comment (viceice) [inline .github/dependabot.yml:1]: nope, will use renovate.

File: pom.xml [hunk 2]
```diff
@@ -42,9 +42,9 @@
       <dependency>
         <groupId>io.jenkins.tools.bom</groupId>
         <artifactId>bom-2.426.x</artifactId>
-        <version>3080.vfa_b_e4a_a_39b_44</version>
-        <type>pom</type>
+        <version>3143.v347db_7c6db_6e</version>
         <scope>import</scope>
+        <type>pom</type>
```
Comment (viceice) [inline pom.xml:47]: please don't change the order

File: pom.xml [hunk 3]
```diff
@@ -5,7 +5,7 @@
   <parent>
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>4.82</version>
+    <version>4.83</version>
```
Comment (viceice) [inline pom.xml:8]: no need to update yet, will be done by renovate when my PR gets merged.

**Review by viceice — COMMENTED** [review]
we need to fix the tests, so please keep changes to minimum.
